wolfcrypt/src/dh.c

```diff
@@ -3167,7 +3176,10 @@ int wc_DhGenerateParams(WC_RNG *rng, int modSz, DhKey *dh)
 
     RESTORE_VECTOR_REGISTERS();
 
-    if (buf != NULL) {
+#ifndef WOLFSSL_NO_MALLOC
+    if (buf != NULL)
+#endif
+    {
         ForceZero(buf, bufSz);
         if (dh != NULL) {
             XFREE(buf, dh->heap, DYNAMIC_TYPE_TMP_BUFFER);
```

Comment:
XFREE on buf which is on the stack with WOLFSSL_NO_MALLOC

Reply:
oh dear, I think you're quite right -- it's only safe to `XFREE()` stack allocations when `XFREE()` is stubbed out as a no-op, as in `-DWOLFSSL_NO_MALLOC -DNO_WOLFSSL_MEMORY`.  will fix.

Reply:
These are all fixed now.  I added a new test scenario to detect them.  There were a bunch of others already there.